core/src/main/java/com/microsoft/applicationinsights/internal/quickpulse/DefaultQuickPulseDataFetcher.java

```diff
@@ -48,28 +49,35 @@
     private String postPrefix;
     private final String sdkVersion;
 
-    public DefaultQuickPulseDataFetcher(ArrayBlockingQueue<HttpPost> sendQueue, TelemetryConfiguration config,
-                                        String instanceName, String quickPulseId) {
-        this(sendQueue, config, null, instanceName, quickPulseId);
+    public DefaultQuickPulseDataFetcher(ArrayBlockingQueue<HttpPost> sendQueue, TelemetryConfiguration config, String machineName,
+                                        String instanceName, String roleName, String quickPulseId) {
+        this(sendQueue, config, null, machineName, instanceName, roleName, quickPulseId);
     }
 
     @Deprecated
-    public DefaultQuickPulseDataFetcher(final ArrayBlockingQueue<HttpPost> sendQueue, final String ikey, final String instanceName, final String quickPulseId) {
-        this(sendQueue, null, ikey, instanceName, quickPulseId);
+    public DefaultQuickPulseDataFetcher(final ArrayBlockingQueue<HttpPost> sendQueue, final String ikey, final String machineName, final String instanceName, final String quickPulseId) {
+        this(sendQueue, null, ikey, machineName, instanceName, quickPulseId);
     }
 
-    private DefaultQuickPulseDataFetcher(ArrayBlockingQueue<HttpPost> sendQueue, TelemetryConfiguration config, String ikey, String instanceName, String quickPulseId) {
+    private DefaultQuickPulseDataFetcher(ArrayBlockingQueue<HttpPost> sendQueue, TelemetryConfiguration config, String ikey, String machineName,
+                                         String instanceName, String roleName, String quickPulseId) {
         this.sendQueue = sendQueue;
         this.config = config;
         this.ikey = ikey;
         sdkVersion = getCurrentSdkVersion();
         final StringBuilder sb = new StringBuilder();
+
+        if (!LocalStringsUtils.isNullOrEmpty(roleName)) {
+            roleName = "\"" + roleName + "\"";
+        }
+
         sb.append("[{");
         formatDocuments(sb);
         sb.append("\"Instance\": \"").append(instanceName).append("\",");
         sb.append("\"InstrumentationKey\": \"").append(ikey).append("\",");
         sb.append("\"InvariantVersion\": 1,");
-        sb.append("\"MachineName\": \"").append(instanceName).append("\",");
+        sb.append("\"MachineName\": \"").append(machineName).append("\",");
+        sb.append("\"RoleName\": ").append(roleName).append(",");
```

Comment:
is it same to remove above conditional:

```
if (!LocalStringsUtils.isNullOrEmpty(roleName)) {
    roleName = "\"" + roleName + "\"";
}
```

and do this (more similar to attributes)?

```suggestion
        sb.append("\"RoleName\": \"").append(roleName).append("\",");
```

Reply:
This is a great question 😃 The problem here is that, without this conditional, if the value of `roleName` is `null` then it will become `roleName: "null"` in the resulting json, as in a string called "null" which is incorrect, we want it to be `roleName: null` so the service interprets this correctly.

Reply:
oh I see, and the others are never null, makes sense 👍